Kafka: Always get the current Metric instance from the metricsSupplier

lbilger opened the pull request:
Fixes #2726

See #2726 for details.

jonatan-ivanov commented on micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java:
```diff
@@ -212,36 +212,42 @@ else if (tags.size() == meterTagsWithCommonTags.size())
 
-    private Meter bindMeter(MeterRegistry registry, Metric metric, String name, Iterable<Tag> tags) {
-        Meter meter = registerMeter(registry, metric, name, tags);
+    private Meter bindMeter(MeterRegistry registry, Supplier<Map<MetricName, ? extends Metric>> metrics, MetricName metricName, String name, Iterable<Tag> tags) {
```
I think improving the variable names a little here and in the other methods would improve readability:
- `metrics` to `metricsSupplier`
- `name` to `meterName`

jonatan-ivanov reviewed (changes requested):
Can we rebase this to 1.6.x so that other LTS versions could be fixed too?

jonatan-ivanov commented:
@lbilger Thanks for the PR, I left a few comments: I think something behaves weirdly (though I tested it after I rebased your changes to `1.6.x`).

lbilger commented:
Hi Jonatan, thanks for having a look! I have rebased to 1.6.x, renamed the variables as requested and added a unit test. 

Can you share your test that showed the weird behavior? `metrics.get().get(metricName)` can return `null` if the metric is no longer present in the Consumer/Producer but `checkAndBindMetrics` has not yet been called since the metric was removed. If that is the case here, I would say it is correct and expected behavior to return NaN. If it is something else, I would have to investigate how it happens.

jonatan-ivanov commented:
The weird behavior was caused by me, in my sample (I used the kafka consumer + producer against a kafka instance), I closed the producer after sending the data so the kafka client removed a bunch of `KafkaMetric` instances so your fix did what it was expected to do, I somehow missed that my producer is closed.
